mlflow/utils/rest_utils.py

```diff
@@ -104,13 +104,23 @@ def http_request(
                 method=method,
                 path=endpoint,
                 headers=extra_headers,
+                # `raw` is passed to `stream` in databricks-sdk. Unfortunately, this is the only way
+                # to get the raw response object.
+                # https://github.com/databricks/databricks-sdk-py/blob/d0d3a933920090b0ef6bacbbd276f1d5886082c2/databricks/sdk/_base_client.py#L292
                 raw=True,
                 query=kwargs.get("params"),
                 body=kwargs.get("json"),
                 files=kwargs.get("files"),
                 data=kwargs.get("data"),
             )
-            return raw_response["contents"]._response
+            if kwargs.get("stream", False):
+                return raw_response["contents"]._response
```

Comment:
There a few calls that use `stream=True`:

https://github.com/mlflow/mlflow/blob/aaca101846abc8457470ec75e8713a4f4ba5bf3b/mlflow/deployments/databricks/__init__.py#L166-L176